### PR TITLE
setup.py, tests/__init__.py: Catch (None, None) locale

### PR DESCRIPTION
**Pull request by corona10**
Sometimes docker images have locale problem.
I found that `locale.getlocale()` can
return (None, None).
So I fixed that when it return (None, None) then setlocale for it.
* setup.py: Catch (None, None) locale
* tests/__init__.py: Catch (None, None) locale

Fixes https://github.com/coala/coala/issues/3906

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,10 @@
 from coalib.misc.BuildManPage import BuildManPage
 
 try:
-    locale.getlocale()
+    lc = locale.getlocale()
+    pf = platform.system()
+    if pf != 'Windows' and lc == (None, None):
```
Comment (Makman2) [inline setup.py]: why checking explicitly for windows?

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import locale
+import platform
+
```
Comment (Makman2) [inline tests/__init__.py]: add a newline ;)

**Comment by Makman2** [comment]
I would maybe split up your commit into two parts, one changing that for installation, one for the tests

\+

Commit message suggestion (when split up):
- `setup.py: Catch (None, None) locale`
- `tests/__init__.py: Catch (None, None) locale`

File: setup.py (hunk 1 above)
Comment (corona10) [inline setup.py]: @Markman2 
I tested on both Linux platform and osx platform.
They worked fine.
but on windows, it looks like has different locale settings.
But I don't have windows machine.

**Comment by supergr35** [comment]
Have you tested if this issue occurs on windows?

**Comment by corona10** [comment]
@supergr35 
I do not test on Windows.
First of all, I don't have Windows machine. 
Second, But it looks like they also return `(None, None)` on windows.
Because when I removed checking platform codes, it was failed on `appveyor/pr` with this message.
```
tests\__init__.py:6: in <module>
    locale.setlocale(locale.LC_ALL, 'C.UTF-8')
C:\Python34\lib\locale.py:592: in setlocale
    return _setlocale(category, locale)
E   locale.Error: unsupported locale setting
```

So I guessed locale problem is not affected on Windows but affected on Linux/Unix OS.

File: setup.py (hunk 1 above)
Comment (Makman2) [inline setup.py]: yeah but you have no reason to exclude it. Windows might too produce `(None, None)` and then the error slips through again :+1:

Comment (Makman2) [inline setup.py]: Nvm just read your post in the PR...

**Comment by Makman2** [comment]
ugh that's weird. As we don't have tests for setup.py I believe, I would test that myself too (or someone else).

--> **Please don't merge until we have tested this properly! Either manually or even better by unit tests!**

**Comment by corona10** [comment]
@Makman2 That's not setup.py. It is tests/__init__.py. Python will execute __init__.py before running test script.

**Comment by corona10** [comment]
@markman2 
Also, this issue is relevant with when we execute `py.test`. 
Because unit test files read `*.rst` file by utf-8 decode.
I also wrote at https://github.com/coala/coala/issues/3906.
as like this. `Also, the same happening when I try to run unit test on local.`

**Comment by corona10** [comment]
@Makman2 And I tested this codes on docker environment and it was succeeded. 
Is there any way to add unit test codes for this environment-sensitive issue?

**Comment by Makman2** [comment]
Yep, we have to add a test file inside `tests/` and reconstruct the conditions for the environment to return `(None, None)` as locale. Then running `setup.py` while installing into a temporary folder. Also we would need to test test-execution itself somehow (without running all tests again)... not that easy, but possible

**Comment by supergr35** [comment]
ping @corona10 , I realize that this is going to take a while, but just checking if you're still working on this.

**Comment by supergr35** [comment]
@Makman2 wouldn't testing test-execution be more of a different issue?

**Comment by corona10** [comment]
@supergr35 Yeah, I looking into this issue. I definitely agree with @Makman2 's opinion. I currently pause for working on this issue for mid-exam. If you can, cherry-pick and proceed it.

**Comment by Makman2** [comment]
@supergr35 thing is we modify the test execution code with something that might break. Although it really is a bit hard, let's try. If it's too hard, I would ack stuff without this "test-for-tests", as the same code for `setup.py` needs definitely tests.

**Comment by supergr35** [comment]
I'm not sure if I've got enough knowledge to accomplish something like that, so I probably shouldn't be the one to cherry pick.

**Comment by jayvdb** [comment]
ack 302bd96 f7da89b

**Comment by jayvdb** [comment]
I think tests-for-tests is not worth the effort here.
Once it is merged, it gets a lot of testing.
If it breaks, we learn something and maybe revert.

**Comment by jayvdb** [comment]
reack a0c9666 d8de173

**Comment by jayvdb** [comment]
@rultor merge
